### PR TITLE
Convert `src/*.js` and `src/*.jsx` to TypeScript 

### PR DESCRIPTION
**Pull request by SimonBrandner**

### CODE REVIEW

File: src/Header.tsx [hunk 1]
```diff
@@ -36,7 +55,18 @@ export function LeftNav({ children, className, hideMobile, ...rest }) {
...
+  children?: ReactNode;
+  className?: string;
+  hideMobile?: string;
+}
+
+export function RightNav({
+  children,
+  className,
+  hideMobile,
+  ...rest
```
Comment (toger5) [inline src/Header.tsx:68]: there is no`rest` in RightNavProps it seems to be the leftover params for a `div` so maybe extend  RightNavProps from `HTMLAttributes<HTMLDivElement>`

File: src/ListBox.jsx [hunk 2]
```diff
@@ -1,50 +0,0 @@
...
-      ref={ref}
-      className={classNames(styles.option, className, {
-        [styles.selected]: isSelected,
-        [styles.focused]: isFocused,
-        [styles.disables]: isDisabled,
-      })}
-    >
-      {item.rendered}
-    </li>
-  );
```
Comment (toger5) [inline src/ListBox.jsx:49]: why does git not understand the rename of the file?

File: src/SequenceDiagramViewerPage.tsx [hunk 3]
```diff
@@ -1,13 +1,20 @@
 import React, { useCallback, useState } from "react";
```
Comment (toger5) [inline src/SequenceDiagramViewerPage.tsx:1]: here we also are missing the copyright

File: src/Tooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import React, {
```
Comment (toger5) [inline src/Tooltip.tsx:1]: here as well (missing copyright)

File: src/UserMenu.tsx [hunk 5]
```diff
@@ -75,12 +85,11 @@ export function UserMenu({
             <UserIcon />
           )}
         </Button>
-        {() => "Profile"}
       </TooltipTrigger>
       {(props) => (
         <Menu {...props} label="User menu" onAction={onAction}>
           {items.map(({ key, icon: Icon, label }) => (
-            <Item key={key} textValue={label} className={styles.menuItem}>
+            <Item key={key} textValue={label}>
```
Comment (toger5) [inline src/UserMenu.tsx:92]: why is the style class not needed here (why did it get removed)?

**Review by toger5 — COMMENTED** [review]
Thanks a lot. Look really good and ticks a lot of files from the non-ts list.
I dont really understand why we move from using the children for the tooltips
```
   const [tooltipTrigger, tooltip] = children;
 ```
 To having it part of the `TooltipTriggerProps`
 ```
 interface TooltipTriggerProps {
  children: ReactElement;
  placement?: Placement;
  delay?: number;
  tooltip: () => string;
}
```

File: src/UserMenu.tsx (hunk 5 above)
Comment (SimonBrandner) [reply src/UserMenu.tsx:92]: `Item` doesn't have a `className` prop

File: src/ListBox.jsx (hunk 2 above)
Comment (SimonBrandner) [reply src/ListBox.jsx:49]: Because the evil overlord want us to suffer :D

**Comment by SimonBrandner** [comment]
> Thanks a lot. Look really good and ticks a lot of files from the non-ts list. I dont really understand why we move from using the children for the tooltips
> 
> ```
>    const [tooltipTrigger, tooltip] = children;
> ```
> 
> To having it part of the `TooltipTriggerProps`
> 
> ```
> interface TooltipTriggerProps {
>  children: ReactElement;
>  placement?: Placement;
>  delay?: number;
>  tooltip: () => string;
> }
> ```

I was having trouble typing it and having it as a prop seems nicer to me somehow - why should we pass down something as a child when it is not a child?

**Review by toger5 — APPROVED** [review]
I see how it is nice to have the tooltip as a prop. Looks fine to me now

**Pull request merged**